### PR TITLE
GH-9617 Add plugin API for GetEmoji method

### PR DESCRIPTION
**Pull request by mojicaj**
#### Summary
Creates the plugin API equivalent of the model/client4.go GetEmoji method

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/9617

#### Checklist
- [x ] Added or updated unit tests (required for all new features)

### CODE REVIEW

**Comment by hmhealey** [comment]
Looks good, except your changes will need to be updated to the latest master since there's some merge conflicts.

File: plugin/api.go [hunk 1]
```diff
@@ -179,6 +179,9 @@ type API interface {
 	// UpdatePost updates a post.
 	UpdatePost(post *model.Post) (*model.Post, *model.AppError)
 
+	// GetEmoji returns a custom emoji based on the emojiId string.
```
Comment (hanzei) [inline plugin/api.go]: Can you document the minimum version required here?

```suggestion
	// GetEmoji returns a custom emoji based on the emojiId string.
	//
	// Minimum server version: 5.6
```

Comment (hmhealey) [reply plugin/api.go]: @jwilander Are we documenting minimum server version for these? I don't know how plugins handle different server versions at this point

Comment (hmhealey) [reply plugin/api.go]: I just saw that a minimum version has been added for other APIs as well, so we should add it here as well

Comment (jwilander) [reply plugin/api.go]: We are, @hanzei is leading the effort :)

**Comment by mojicaj** [comment]
@hanzei Sorry for the delayed response, it has been hectic at work.  I will try to get these changes made this week.

**Comment by mojicaj** [comment]
conflicts resolved and added the server version

**Comment by hmhealey** [comment]
Looks good. I'll merge it now before any more conflicts come up 😛

**Pull request merged**